hnn_core/network.py

```diff
@@ -14,6 +14,54 @@
 from .params import create_pext
 
 
+def read_spikes(fname, gid_dict=None):
+    """Read spiking activity from a collection of spike trial
+    files.
+
+    Parameters
+    ----------
+    fname : str
+        Wildcard expression (e.g., '<pathname>/spk_*.txt') of the
+        path to the spike file(s).
+    gid_dict : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, each spike file must contain
+        a 3rd column for spike type.
+
+    Returns
+    ----------
+    spikes : Spikes object
+        An instance of the Spikes object.
+    """
+    from glob import glob
```

Comment:
You should always prefer global imports when writing a package. The reason is that the import will happen every time you call this function. So imagine if there is a function which calls this function 100 times, you will have `glob` imported 100 times and each of those imports will take some time. The only exceptions to this rule are:

1) matplotlib imports. We do these nested inside functions because of some weird bug. I don't exactly know the reason.
2) circular imports. If you file A containing function a1 calls file B containing function b, and then function b calls a function a2 in file A. Then you would need to nest the function. Usually this is an indicator of bad code structure though.
3) partial dependencies: if you want to have a partial dependency on a package so that it's imported only by users who want to use an exotic function. This should be used sparingly.